sql/core/src/test/resources/sql-tests/results/random.sql.out

```diff
@@ -361,6 +361,46 @@ struct<result:string>
 ceV0PXaR2I
 
 
+-- !query
+SELECT randstr(CAST(10 AS TINYINT), 0) AS result
+-- !query schema
+struct<result:string>
+-- !query output
+ceV0PXaR2I
+
+
+-- !query
+SELECT randstr(CAST(10 AS BIGINT), 0) AS result
+-- !query schema
+struct<result:string>
+-- !query output
+ceV0PXaR2I
+
+
+-- !query
+SELECT randstr(1.0F, 0) AS result
+-- !query schema
+struct<result:string>
+-- !query output
+c
+
+
+-- !query
+SELECT randstr(1.0D, 0) AS result
```

Comment:
and truncation (from double to int) is allowed?

Reply:
Yes, for the first argument representing the length of the random string to generate. This is per request from @srielau in https://issues.apache.org/jira/browse/SPARK-50560. 

No, for the second argument representing the random seed to use. This behavior is consistent with the existing `RAND` and `RANDN` functions which specify that the random seed must be `IntegerType` or `LongType` (but may be positive, zero, or negative).

I added a few tests to exercise this.